src/command_exec_utils.rs

```diff
@@ -1,33 +1,47 @@
 use serde::{Deserialize, Serialize};
-use serde_json::{Map, Value};
-use std::process::Command;
-use std::{error::Error};
+use std::collections::HashMap;
+use std::error::Error;
+use std::path::{Path, PathBuf};
+use std::process::{exit, Child, Command, Stdio};
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct Script {
     pub(crate) script: String,
 }
 
-pub fn exec_scripts(command: &str, projects: Vec<String>) -> Result<(), Box<dyn Error>> {
-   // let result: Map<String, Value> = Map::new();
-   let output = Command::new("npm").arg("run").arg(command).output()?;
+pub fn exec_scripts<'a>(
+    command: &str,
+    projects: Vec<&'a str>,
+) -> Result<HashMap<&'a str, String>, Box<dyn Error>> {
+    let mut ids: HashMap<&str, String> = HashMap::new();
 
-    println!("{:?}", projects);
-    if !output.status.success() {
-        println!("Command executed with failing error code");
+    for p in projects {
+        let mut child = exec_script(command, p);
+        ids.insert(p, child.id().to_string());
```

Comment:
storing the child process id for future termination by demand - basically I thought of 3 approaches:
1. store a reference to the process itself and use its kill() method - faced borrowing issues
2. store a reference to the process kill() method for future execution, tried to use the Box pointer but faced borrowing issues as well.
3. store the process id and use an external kill_by_id function - no issues with this functional implementation.. but I would like to figure out the first 2 options **